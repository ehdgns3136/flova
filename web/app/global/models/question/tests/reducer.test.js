
import { fromJS } from 'immutable';
import questionReducer from '../reducer';
import {
  openFollowerModal,
  closeFollowerModal,
  getQuestionFollowersRequest,
  getQuestionFollowersSuccess,
  getQuestionFollowersFailure,
} from '../actions';
import { MOCK_USERS } from './mocks/index';


describe('questionReducer', () => {
  it('should return the initial state', () => {
    expect(questionReducer(undefined, {})).toMatchSnapshot();
  });

  it('should handle openFollowerModal action correctly', () => {
    const state1 = fromJS({
      isFollowerModalOpened: false,
    });

    const stateResult1 = fromJS({
      isFollowerModalOpened: true,
    });

    expect(questionReducer(state1, openFollowerModal())).toEqual(stateResult1);
  });

  it('should handle closeFollowerModal action correctly', () => {
    const state1 = fromJS({
      isFollowerModalOpened: true,
    });

    const stateResult1 = fromJS({
      isFollowerModalOpened: false,
    });

    expect(questionReducer(state1, closeFollowerModal())).toEqual(stateResult1);
  });

  it('should handle getQuestionFollowersRequest action correctly', () => {
    const state1 = fromJS({
      followers: null,
      isFollowersLoading: false,
    });

    const stateResult1 = fromJS({
      followers: null,
      isFollowersLoading: true,
    });

    expect(questionReducer(state1, getQuestionFollowersRequest())).toEqual(stateResult1);
  });

  it('should handle getQuestionFollowersSuccess action correctly', () => {
    const state1 = fromJS({
      followers: null,
      isFollowersLoading: true,
    });

    const stateResult1 = fromJS({
      followers: [
        1,
        2,
        3,
      ],
      isFollowersLoading: false,
    });

    expect(questionReducer(state1, getQuestionFollowersSuccess(MOCK_USERS))).toEqual(stateResult1);
  });

  it('should handle getQuestionFollowersFailure action correctly', () => {
    const state1 = fromJS({
      followers: [
        1,
        2,
        3,
      ],
      isFollowersLoading: true,
    });

    const stateResult1 = fromJS({
      followers: null,
      isFollowersLoading: false,
    });

    expect(questionReducer(state1, getQuestionFollowersFailure())).toEqual(stateResult1);
  });
});
