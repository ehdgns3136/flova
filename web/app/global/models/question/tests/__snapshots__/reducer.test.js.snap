exports[`questionReducer should return the initial state 1`] = `
Object {
  "followers": null,
  "initialContent": null,
  "initialTitle": null,
  "initialTopics": null,
  "isFollowerModalOpened": false,
  "isFollowersLoading": false,
  "isQuestionEditFormModalOpened": false,
  "isQuestionFormModalOpened": false,
  "isQuestionUploadError": false,
  "isQuestionUploading": false,
  "questionID": null,
}
`;
